automation_tools/__init__.py

```diff
@@ -1465,3 +1469,178 @@ def setenforce(mode):
+    """
+    if (instance_name or image_name or flavor_name or ssh_key) is None:
+        raise ValueError(
+            'instance_name/image_name/flavor_name/ssh_key cannot be None.')
+    network_name = 'satellite-jenkins'
+    clientObj = client.Client(
+        username='satellite-jenkins',
+        api_key='dog8code',
+        auth_url='http://qeos.lab.eng.rdu2.redhat.com:5000/v2.0',
+        project_id='satellite-jenkins'
+    )
+    # Validate openstack credentials.
+    clientObj.authenticate()
+    # Validate ssh_key is added into openstack project
+    clientObj.keypairs.find(name=ssh_key)
+    image = clientObj.images.find(name=image_name)
+    flavor = clientObj.flavors.find(name=flavor_name)
+    network = clientObj.networks.find(label=network_name)
+    floating_ip = clientObj.floating_ips.create(
+        clientObj.floating_ip_pools.list()[0].name
+    )
+    # Create instance from the given parameters
+    instance = clientObj.servers.create(
+        name=instance_name,
+        image=image.id,
+        flavor=flavor.id,
+        key_name=ssh_key,
+        network=network.id
+    )
+    # Average time considered to wait for instance to get in \
+    # ACTIVE running state, for assigning floating ip
+    time.sleep(300)
+    instance.add_floating_ip(floating_ip)
+    # Wait till DNS resolves the IP
+    time.sleep(600)
+    # Getting Hostname from IP
+    env['INSTANCE_HOST'] = get_hostname_from_ip(str(floating_ip.ip))
+
+
+def delete_openstack_instance(instance_name=None):
+    """Deletes openstack Instance.
+
+    :param str instance_name: Openstack instance name to delete.
+    """
+    if instance_name is None:
+        raise ValueError('Instance name cannot be None.')
+    clientObj = client.Client(
+        username='satellite-jenkins',
+        api_key='dog8code',
+        auth_url='http://qeos.lab.eng.rdu2.redhat.com:5000/v2.0',
+        project_id='satellite-jenkins'
+    )
+    try:
+        instance = clientObj.servers.find(name=instance_name)
+    except novaclient.exceptions.NotFound:
+        print('Instance {0} in not listed in Openstack project.'.format(
+            instance_name
+        ))
+        return
+    instance.delete()
+
+
+def satellite_upgrade(admin_password=None):
+    """Upgrades satellite from already created Openstack image \
+    of old Satellite version to latest Satellite version compose.
+
+    The following environment variables affect this command:
+
+    ADMIN_PASSWORD
+        Optional, defaults to 'changeme'. Foreman admin password.
+    BASE_URL
+        URL for the compose repository.
+    """
+    base_url = os.environ.get('BASE_URL')
+    if base_url is None:
+        print('The BASE_URL environment variable should be defined')
+        sys.exit(1)
+    if admin_password is None:
+        admin_password = os.environ.get('ADMIN_PASSWORD', 'changeme')
+    # Removing rhel-released and rhel-optional repo
+    run('rm -rf /etc/yum.repos.d/rhel-optional.repo')
```

Comment:
`run('rm -rf /etc/yum.repos.d/rhel-{optional,released}.repo')` should remove both with one command.